servers/graphql-kotlin-spring-server/src/main/kotlin/com/expediagroup/graphql/spring/execution/ApolloSubscriptionHooks.kt

```diff
@@ -25,30 +25,37 @@ import reactor.core.publisher.Mono
  * https://www.apollographql.com/docs/graphql-subscriptions/lifecycle-events/
  */
 interface ApolloSubscriptionHooks {
+
     /**
      * Allows validation of connectionParams prior to starting the connection.
-     * You can reject the connection by throwing an exception
+     * You can reject the connection by throwing an exception.
+     * If you need to forward state to execution, update and return the [GraphQLContext].
      */
-    fun onConnect(connectionParams: Map<String, String>, session: WebSocketSession, graphQLContext: Any?): Mono<Unit> = mono {}
+    fun onConnect(
+        connectionParams: Map<String, String>,
+        session: WebSocketSession,
+        graphQLContext: GraphQLContext
+    ): Mono<GraphQLContext> = Mono.just(graphQLContext)
 
     /**
-     * Called when the client executes a GraphQL operation
+     * Called when the client executes a GraphQL operation.
+     * The context can not be updated here, it is read only.
      */
     fun onOperation(
         operationMessage: SubscriptionOperationMessage,
         session: WebSocketSession,
-        graphQLContext: Any?
-    ): Mono<Unit> = mono { }
+        graphQLContext: GraphQLContext
+    ): Unit = Unit
```

Comment:
why the change from `Mono` to `Unit`?

Reply:
Because it drastically simplifies the subscription logic. Since we couldn't return anything from these hooks anyway we can just make them `Unit` functions that just get called in the life cycle.

Reply:
By not requiring mono we dont need to have all the combiners in the subscription handler to combine the hooks monos with the real data fluxs